src/core/aws-net-websocket-registry.adb

```diff
@@ -614,6 +679,21 @@ package body AWS.Net.WebSocket.Registry is
          end;
       end Get_Socket;
 
+      ----------------
+      -- Get_Socket --
+      ----------------
+
+      procedure Get_Socket (Id : UID; WebSocket : out Object_Class) is
+      begin
+         if Registered.Contains (Id) then
+            Sending.Insert (Id);
+
+            WebSocket := Registered (Id);
```

Comment:
There are two lookups in `Registered` here, better to get a `Cursor`  and check `Has_Element` on it.